app/javascript/dashboard/components/widgets/conversation/WhatsappTemplates/TemplatesPicker.vue

```diff
@@ -39,9 +32,31 @@ export default {
     },
   },
   methods: {
-    getTemplatebody(template) {
-      return template.components.find(component => component.type === 'BODY')
-        .text;
+    getTemplateBody(template) {
+      return (
+        template.components.find(component => component.type === 'BODY')
+          ?.text || ''
+      );
+    },
+    getTemplateHeader(template) {
+      return template.components.find(component => component.type === 'HEADER');
+    },
+    getTemplateFooter(template) {
+      return template.components.find(component => component.type === 'FOOTER');
+    },
+    getTemplateButtons(template) {
+      return template.components.find(
+        component => component.type === 'BUTTONS'
+      );
+    },
```

Comment:
Let's create a common helper method for this logic in `templateHelper`. You can use the same helper here as well. 

1. https://github.com/chatwoot/chatwoot/pull/12117/files#diff-e6421f31fd1b5d4c455e5d06e9f09c5d7039a5b05c328dd598c56032962e6e4bR46-R54
2. https://github.com/chatwoot/chatwoot/pull/12117/files#diff-38a62600931c2509072e530ccb1b4cc732ee29719155456cb1b192ca71863834R20-R25

```
const findComponentByType = (template, type) =>
  template.components?.find(component => component.type === type);
```